app/models/validations/financial_validations.rb

```diff
@@ -58,4 +58,48 @@ def validate_tshortfall(record)
       record.errors.add :tshortfall, I18n.t("validations.financial.hbrentshortfall.outstanding_no_benefits")
     end
   end
+
+  SCHARGE_RANGES = { this_landlord_general_needs:
+  { min: 0, max: 55, error: I18n.t("validations.financial.rent.scharge.this_landlord.general_needs") },
+                     this_landlord_supported_housing:
+  { min: 0, max: 280, error: I18n.t("validations.financial.rent.scharge.this_landlord.supported_housing") },
+                     other_landlord_general_needs:
+  { min: 0, max: 45, error: I18n.t("validations.financial.rent.scharge.other_landlord.general_needs") },
+                     other_landlord_supported_housing:
+  { min: 0, max: 165, error: I18n.t("validations.financial.rent.scharge.other_landlord.supported_housing") } }.freeze
+
+  def validate_rent_amount(record)
+    if record.brent.present? && record.tshortfall.present? && record.brent < record.tshortfall * 2
+      record.errors.add :brent, I18n.t("validations.financial.rent.less_than_double_shortfall", tshortfall: record.tshortfall * 2)
+      record.errors.add :tshortfall, I18n.t("validations.financial.tshortfall.more_than_rent")
+    end
+
+    validate_scharge(record)
+  end
+
+private
+
+  def validate_scharge(record)
+    scharge_range = if record.this_landlord?
+                      if record.is_general_needs?
+                        SCHARGE_RANGES[:this_landlord_general_needs]
+                      elsif record.is_supported_housing?
+                        SCHARGE_RANGES[:this_landlord_supported_housing]
+                      end
+                    elsif record.other_landlord?
+                      if record.is_general_needs?
+                        SCHARGE_RANGES[:other_landlord_general_needs]
+                      elsif record.is_supported_housing?
+                        SCHARGE_RANGES[:other_landlord_supported_housing]
+                      end
+                    end
+
+    if scharge_range.present? && !weekly_value_in_range(record, "scharge", scharge_range[:min], scharge_range[:max])
+      record.errors.add :scharge, scharge_range[:error]
+    end
+  end
```

Comment:
@baarkerlounger this is not ideal and I feel like I'm missing a very basic refactor somewhere here, any suggestions?

Reply:
If you make it a nested hash you could do a direct look up. Something like:

```
SCHARGE_RANGES = {
  this_landlord: {
    general_needs: {
      range: 0..55,
      error: I18n.t("validations.financial.rent.scharge.this_landlord.general_needs")
    },
    supported_housing: {
      range: 0..200,
      error: I18n.t("validations.financial.rent.scharge.this_landlord.supported_housing")
    }
  },
  other_landlord: {
    general_needs: {
      range: 0..45,
      error: I18n.t("validations.financial.rent.scharge.other_landlord.general_needs")
    },
    supported_housing: {
      range: 0..165,
      error: I18n.t("validations.financial.rent.scharge.other_landlord.supported_housing")
    }
  }
}.freeze

SCHARGE_RANGES.dig(:this_landlord, :general_needs)
```

Reply:
Hmm, that still requires writing out all the if statements for is_this_landlord and is_general_needs tho?